framework/src/seclink/seclink.c

```diff
@@ -55,26 +55,35 @@
 
 #ifdef LINUX
 extern int sl_post_msg(int fd, int cmd, unsigned long arg);
+#define ioctl sl_post_msg
+#endif
 
 #define SL_CALL(hnd, code, param)										\
 	do {																\
-		int i_res = sl_post_msg(hnd->fd, code, (unsigned long)((uintptr_t)&param)); \
+		int i_res = ioctl(hnd->fd, code, (unsigned long)((uintptr_t)&param)); \
 		if (i_res < 0) {												\
 			SL_ERR(i_res);												\
 			return SECLINK_ERROR;										\
 		}																\
 	} while (0)
-#else
-#define SL_CALL(hnd, code, param)										\
+
+#define SL_CALL2(hnd, code, param, error)								\
 	do {																\
 		int i_res = ioctl(hnd->fd, code, (unsigned long)((uintptr_t)&param)); \
 		if (i_res < 0) {												\
 			SL_ERR(i_res);												\
+			error;														\
```

Comment:
perhaps give it a meaningful name as well, such as `restore_on_error` or `error_cleanup`?

Reply:
I think error is understandable.